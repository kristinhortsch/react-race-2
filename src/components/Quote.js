import React from 'react';
import PropTypes from 'prop-types';

export default function Quote({ quote, onClick }) {
  return (
    <>
      <h1>Quote</h1>
      <p>{quote.text}</p>
      <button onClick={onClick}>New Quote</button>
    </>
  );
}

Quote.propTypes = {
  quote: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};
