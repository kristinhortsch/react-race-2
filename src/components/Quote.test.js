import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('Quote', () => {
  it('matches a snapshot', () => {
    const quote = {
      text: 'some quote'
    };

    const onClick = jest.fn();
    const tree = renderer.create(
      <Quote onClick={onClick} quote={quote}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
