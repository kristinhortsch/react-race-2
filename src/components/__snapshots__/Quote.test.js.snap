// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote matches a snapshot 1`] = `
Array [
  <h1>
    Quote
  </h1>,
  <p>
    some quote
  </p>,
  <button
    onClick={[MockFunction]}
  >
    New Quote
  </button>,
]
`;
